### PR TITLE
Frontend auth refactor

### LINKED ISSUE #37: Implement authentication token handling on the client

### PR DESCRIPTION
**Pull request by aleksakojadinovic**
_NOTE: THE HUGE 66 FILE DIFF IS DUE TO APPLYING CODE FORMATTER, MOST OF IT IS IRRELEVANT_

- Implemented the axios interceptor
- Refactor login and register thunks
    - Register was done using ```createAsyncThunk```, but most of the work was done manually, so I made it into a regular thunk
    - I've also converted the login into a regular thunk since I need to perform side-effects (local storage). This could have also been done using ```createListenerMiddleware``` but it was waaay to difficult and we're short on time
- I've wrapped the entire App into ```WithLocalStorageFetcher``` HOC which just reads tokens from the local storage and copies them to redux. There is no live refreshing now, however I think I can manage that as well since passing around store instance is easier than I thought.
- Added .prettierrc
    - Granted, this should have been done in a separate PR. However if I had not done it now it would have probably taken me 5 times longer to finish this feature
- I've left an example on the HomePage if you want to test this out. There's a Get All Users button which just calls that API. I've set the tokens to last only 5 seconds. The best way to test it is to open the console and look at the logs/network

Closes #37

### CODE REVIEW

File: app/src/axios/instance.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    (error) => {
+        if (!error.response || error.response.status !== 401) {
+            return Promise.reject(error);
+        }
+        const originalRequest = error.config;
+        if (!error.response.headers['is-token-expired']) {
+            // Meaning the request was unauthorized or some other error occurred
+            console.log('[response interceptor]: unauthorized, skip');
+            return Promise.reject(error);
+        }
```
Comment (VladimirV99) [inline app/src/axios/instance.ts:51]: Mozes li da spojis ova dva if?

File: app/src/axios/instance.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+const axiosAuthInstance = axios.create();
+
+const refreshSession = () => {
+    const refreshToken = window.localStorage.getItem('refreshToken');
+    const userEncoded = window.localStorage.getItem('user');
+    const email = JSON.parse(userEncoded || '')?.email;
+
+    return axios.post(
+        'http://localhost:5106/api/v1/Authentication/RefreshSession',
```
Comment (VladimirV99) [inline app/src/axios/instance.ts:13]: Adresa servera treba da se izvuce u promenljivu

File: app/src/redux/auth/reducers/index.ts [hunk 3]
```diff
@@ -1,37 +1,59 @@
...
+    });
+    builder.addCase(setTokensAndUser, (state, action) => {
+        state.accessToken = action.payload.accessToken;
+        state.refreshToken = action.payload.refreshToken;
+        state.user = action.payload.user;
+    });
+    builder.addCase(clearTokensAndUser, (state) => {
+        state.accessToken = null;
+        state.refreshToken = null;
+        state.user = null;
```
Comment (VladimirV99) [inline app/src/redux/auth/reducers/index.ts:55]: Jel ovo radi isto sto i logoutCustomer?

File: app/src/axios/instance.ts (hunk 2 above)
Comment (aleksakojadinovic) [reply app/src/axios/instance.ts:13]: ako uvodimo ona API gateway uskoro ionako cemo da menjamo skroz kako se URL-ovi cuvaju tako da bih iskreno ostavio za sad

File: app/src/redux/auth/reducers/index.ts (hunk 3 above)
Comment (aleksakojadinovic) [reply app/src/redux/auth/reducers/index.ts:55]: da realno

**Review by VladimirV99 — APPROVED** [review]
Sve deluje ok, samo ne zaboravi da vratis podesavanja za identity server i da obrises debug ispis u konzoli

**Pull request merged**